File: edupath-android/app/src/main/java/com/edupath/data/api/dto/AuthDto.kt
package com.edupath.data.api.dto

import com.squareup.moshi.Json
import com.squareup.moshi.JsonClass

/**
 * Request body for user login.
 *
 * @property username The user's username or email
 * @property password The user's password
 */
@JsonClass(generateAdapter = true)
data class LoginRequest(
    val username: String,
    val password: String
)

/**
 * Response containing JWT tokens after successful authentication.
 *
 * @property access The access token for API requests (short-lived)
 * @property refresh The refresh token for obtaining new access tokens (long-lived)
 */
@JsonClass(generateAdapter = true)
data class AuthResponse(
    val access: String,
    val refresh: String? = null
)

/**
 * Request body for refreshing an expired access token.
 *
 * @property refresh The refresh token to exchange for a new access token
 */
@JsonClass(generateAdapter = true)
data class RefreshTokenRequest(
    val refresh: String
)

/**
 * Generic response containing a message.
 *
 * Used for success/error responses from various endpoints.
 *
 * @property message The response message
 */
@JsonClass(generateAdapter = true)
data class MessageResponse(
    val message: String
)

File: edupath-ios/EduPath/Data/DTOs/AuthDTO.swift
import Foundation

/// Login request body.
struct LoginRequest: Encodable {
    let username: String
    let password: String
}

/// Token refresh request body.
struct RefreshTokenRequest: Encodable {
    let refresh: String
}

/// Authentication response containing JWT tokens.
struct AuthResponse: Decodable {
    let access: String
    let refresh: String?
}

/// User data from API.
struct UserDTO: Decodable {
    let id: Int
    let username: String
    let email: String
    let firstName: String
    let lastName: String
    let isStaff: Bool
    let isSuperuser: Bool
    let dateJoined: Date?

    func toDomain() -> User {
        User(
            id: id,
            username: username,
            email: email,
            firstName: firstName,
            lastName: lastName,
            isStaff: isStaff,
            isSuperuser: isSuperuser,
            dateJoined: dateJoined
        )
    }
}

/// User profile data from API.
struct UserProfileDTO: Decodable {
    let id: Int
    let user: Int
    let avatar: String?
    let bio: String?
    let phone: String?
    let website: String?
    let linkedinUrl: String?
    let twitterUrl: String?
    let githubUrl: String?
    let emailNotifications: Bool

    func toDomain() -> UserProfile {
        UserProfile(
            id: id,
            userId: user,
            avatarURL: avatar.flatMap { URL(string: $0) },
            bio: bio ?? "",
            phone: phone ?? "",
            website: website.flatMap { URL(string: $0) },
            linkedinURL: linkedinUrl.flatMap { URL(string: $0) },
            twitterURL: twitterUrl.flatMap { URL(string: $0) },
            githubURL: githubUrl.flatMap { URL(string: $0) },
            emailNotifications: emailNotifications
        )
    }
}
